tests/api2/plugin.py

```diff
@@ -0,0 +1,379 @@
+    global transmission_jail, results
+    results = GET("/jail/id/transmission")
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    transmission_jail = results.json()
+
+
+@to_skip
+@pytest.mark.parametrize('prop', plugin_objects)
+def test_16_verify_transmission_plugin_value_with_jail_value_of_(prop):
+    assert transmission_jail[prop] == transmission_plugin[prop], results.text
+
+
+@to_skip
+def test_17_get_list_of_available_plugins_without_cache():
+    global JOB_ID
+    payload = {
+        "plugin_repository": repos_url,
+        "cache": False
+    }
+    results = POST('/plugin/available/', payload)
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), int), results.text
+    JOB_ID = results.json()
+
+
+@to_skip
+def test_18_verify_list_of_available_plugins_job_id_is_successfull():
+    global job_results
+    job_status = wait_on_job(JOB_ID)
+    assert job_status['state'] == 'SUCCESS', job_status['results']
+    job_results = job_status['results']
+
+
+@to_skip
+@pytest.mark.parametrize('plugin', plugin_list)
+def test_19_verify_available_plugin_without_cache_(plugin):
+    assert isinstance(job_results['result'], list), str(job_results)
+    for plugin_info in job_results['result']:
+        if plugin in plugin_info['plugin']:
+            assert plugin in plugin_info['plugin'], str(job_results)
+            assert isinstance(plugin_info, dict), str(job_results)
+
+
+@to_skip
+def test_20_stop_transmission_jail():
+    global JOB_ID
+    payload = {
+        "jail": "transmission",
+        "force": True
+    }
+    results = POST('/jail/stop/', payload)
+    assert results.status_code == 200, results.text
+    JOB_ID = results.json()
+
+
+@to_skip
+def test_21_wait_for_transmission_plugin_to_be_down():
+    job_status = wait_on_job(JOB_ID)
+    assert job_status['state'] == 'SUCCESS', job_status['results']
+    results = GET('/plugin/id/transmission/')
+    assert results.json()['state'] == 'down', results.text
+
+
+@to_skip
+def test_22_start_transmission_jail():
+    global JOB_ID
+    payload = "transmission"
+    results = POST('/jail/start/', payload)
+    assert results.status_code == 200, results.text
+    JOB_ID = results.json()
+
+
+@to_skip
+def test_23_wait_for_transmission_plugin_to_be_up():
+    job_status = wait_on_job(JOB_ID)
+    assert job_status['state'] == 'SUCCESS', job_status['results']
+    results = GET('/plugin/id/transmission/')
+    assert results.json()['state'] == 'up', results.text
+
+
+@to_skip
+def test_24_stop_transmission_jail_before_deleteing():
+    global JOB_ID
+    payload = {
+        "jail": "transmission",
+        "force": True
+    }
+    results = POST('/jail/stop/', payload)
+    assert results.status_code == 200, results.text
+    JOB_ID = results.json()
+
+
+@to_skip
+def test_25_wait_for_transmission_plugin_to_be_down():
+    job_status = wait_on_job(JOB_ID)
```

Comment:
We don't want to wait for `10` minutes for stopping a plugin. That unnecessarily adds a lot of time to the tests. We should allow a `max_timeout` parameter for `wait_on_job` function which it can use instead and it can default to `600`.